package/src/index/scrape.rs

```diff
@@ -0,0 +1,69 @@
+//! Tools for updating an index from a git repository.
+
+use gix::ObjectId;
+use nickel_lang_git::Spec;
+use tempfile::tempdir;
+
+use crate::{
+    error::{Error, IoResultExt},
+    manifest::MANIFEST_NAME,
+    ManifestFile,
+};
+
+use super::{Id, Package, PreciseId};
+
+/// Fetch a package from the specified place, and figure out what its index
+/// entry should look like.
+///
+/// Question: should we allow index packages to come from a _subdirectory_ of
+/// a git repo, or should they always need to be at the root? Currently we only
+/// allow them at the root, the motivation being that if the git repo backing
+/// the index package only contains the package itself, then it will be a
+/// smaller download. On the other hand, we allow git dependencies to live in
```

Comment:
I'm not sure to understand the motivation: if we allow subdirs, we can still allow being it at the root, without paying additional price, right? Or put differently: is allowing subdirs affecting negatively the optimistic workflow of "package is the whole repo"? Or is more around forcing people to do it to ensure the index is more performant? I think we can leave that for future work, but the mono repo case surely sounds plausible.

Reply:
It was mainly about avoiding the performance surprise of depending on an index package and then downloading a big monorepo.

But also, when a rust crate is published from a monorepo the packaging step extracts just the crate that you're publishing. So I thought we could have some similar tooling, where when you package from a monorepo the tooling extracts just the part you want to publish into a smaller repo. (Or maybe even into a compressed tarball)

Reply:
I see. So the answer is that you'd prefer to do it properly, rather than blindly downloading the monorepo, and that is possible but just require some work?

Reply:
I mean, it's less work on the package consumption side (one less case to handle, no need to think about maybe deduplicating the monorepos...) but a bit more work on the package creation side.